src/wwElement.vue

```diff
@@ -1,535 +1,789 @@
-        :locale="currentLang"
-        :class="content.themeColor"
-        :style="{ ...customThemeStyle, ...calendarsStyle, ...categoriesStyle }"
-        :time="!content.enableTimelessMode"
-        :timeStep="content.timestep"
-        :timeFrom="content.timeStart * 60"
-        :timeTo="content.timeEnd * 60"
-        :startWeekOnSunday="content.startWeekOnSunday"
-        :hideWeekdays="content.hideWeekdays"
-        :twelveHour="content.twelveHour"
-        :small="content.daySize === 'small'"
-        :xsmall="content.daySize === 'xsmall'"
-        :watchRealTime="content.watchRealTime"
-        :todayButton="content.todayButton"
-        v-model:selectedDate="internalSelectedDate"
-        @event-click="handleEventClick"
-        @cell-click="handleCellClick"
-        @view-change="handleViewChange"
-        @ready="handleReady"
-    />
+    <div class="fullcalendar-wrapper modern-buttons" :class="{ 'dark-mode': isDarkMode }" :style="calendarStyles">
+        <FullCalendar ref="fullCalendarRef" :options="calendarOptions">
+            <template v-slot:noEventsContent>
+                {{ noEventsContentText }}
+            </template>
+        </FullCalendar>
+    </div>
 </template>
 
 <script>
-import { computed } from 'vue';
-import VueCal from 'vue-cal';
-import 'vue-cal/dist/vuecal.css';
-
-import * as en from 'vue-cal/dist/i18n/en.es.js';
-import * as fr from 'vue-cal/dist/i18n/fr.es.js';
-import * as es from 'vue-cal/dist/i18n/es.es.js';
-import * as de from 'vue-cal/dist/i18n/de.es.js';
-import * as nl from 'vue-cal/dist/i18n/nl.es.js';
-import * as it from 'vue-cal/dist/i18n/it.es.js';
-import * as pt from 'vue-cal/dist/i18n/pt-br.es.js';
-import * as ar from 'vue-cal/dist/i18n/ar.es.js';
-
-const locales = {
-    ar: ar,
-    en: en,
-    fr: fr,
-    es: es,
-    de: de,
-    nl: nl,
-    it: it,
-    'pt-br': pt,
-};
-
-const UNBOUNCED_DELAY_IN_MS = 200;
+import { ref, computed, watch } from 'vue';
+import FullCalendar from '@fullcalendar/vue3';
+import dayGridPlugin from '@fullcalendar/daygrid';
+import timeGridPlugin from '@fullcalendar/timegrid';
+import listPlugin from '@fullcalendar/list';
+import interactionPlugin from '@fullcalendar/interaction';
+import multiMonthPlugin from '@fullcalendar/multimonth';
 
 export default {
-    components: { VueCal },
+    components: {
+        FullCalendar,
+    },
     props: {
-        uid: { type: String, required: true },
         content: { type: Object, required: true },
+        uid: { type: String, required: true },
         /* wwEditor:start */
         wwEditorState: { type: Object, required: true },
         /* wwEditor:end */
     },
-    emits: ['trigger-event', 'update:content:effect'],
-    setup(props) {
-        const { value: selectedDate, setValue: setSelectedDate } = wwLib.wwVariable.useComponentVariable({
+    emits: ['trigger-event'],
+    setup(props, { emit }) {
+        const fullCalendarRef = ref(null);

```

Comment:
avec la nouvelle version de vue tu peux utiliser useTemplateRef

Reply:
C'est quoi la différence ?

Reply:
en typescript ça type, ici c'est surtout une histoire de doc/intention => on voit de suite que c'est une référence template. C'est du nice to have